src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -3389,28 +3383,26 @@ def cli_cosmosdb_table_restore(cmd,
         if restorable_database_account is None:
             raise CLIError("Cannot find a database account with name {} that is online".format(account_name))
 
-        latest_restore_time = datetime.datetime.utcfromtimestamp(0)
         try:
             from azure.cli.command_modules.cosmosdb._client_factory import cf_restorable_tables
             restorable_tables_client = cf_restorable_tables(cmd.cli_ctx, [])
             restorable_tables = restorable_tables_client.list(
                 restorable_database_account.location,
                 restorable_database_account.name)
 
-            for restorable_table in restorable_tables:
-                resource = restorable_table.resource
-                event_timestamp = datetime.datetime.strptime(resource.event_timestamp, "%Y-%m-%dT%H:%M:%SZ")
-                if resource.operation_type == "Delete" and latest_restore_time < event_timestamp and resource.owner_id == table_name:
-                    latest_restore_time = event_timestamp
+            latest_table_delete_time, latest_table_create_or_recreate_time, table_rid = process_restorable_databases(restorable_tables, table_name)  # pylint: disable=unused-variable
 
-            if (latest_restore_time == datetime.datetime.utcfromtimestamp(0)):
-                raise CLIError("Cannot find a deleted table with name {} in this account with name {} that is online in location {}".format(table_name, account_name, restorable_database_account.location))
+            # Table is alive if create or recreate timestamp is later than latest delete timestamp
+            table_alive = latest_table_create_or_recreate_time > latest_table_delete_time or latest_table_delete_time == datetime.datetime.max
+
+            if (table_alive):
```

Comment:
please correct the error message and if syntax similar to sql container or others.

Reply:
thanks for pointing out @amisi01. have updated code.